### PR TITLE
fix(FEC-8826): fullscreen implementation moved to core

### PR DESCRIPTION
**Pull request by Yuvalke**
### Description of the Changes

Element parameter to open in fullscreen and new method for registeration to fullscreen events

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/kaltura-player.js [hunk 1]
```diff
@@ -260,6 +260,10 @@ class KalturaPlayer extends FakeEventTarget {
     return this._localPlayer.isFullscreen();
   }
 
+  registerFullScreenEvents(EventHandler: Function): void {
```
Comment (yairans) [inline src/kaltura-player.js]: not needed. remove.

File: src/kaltura-player.js [hunk 2]
```diff
@@ -268,8 +272,8 @@ class KalturaPlayer extends FakeEventTarget {
     this._localPlayer.notifyExitFullscreen();
   }
 
-  enterFullscreen(): void {
-    this._localPlayer.enterFullscreen();
+  enterFullscreen(fullScreenElement: HTMLElement): void {
```
Comment (yairans) [inline src/kaltura-player.js]:
```suggestion
  enterFullscreen(fullScreenElement: ?HTMLElement): void {
```

**Comment by yairans** [comment]
@Yuvalke update `supportLegacyOptions` test to cover your change

File: src/common/utils/setup-helpers.js [hunk 3]
```diff
@@ -342,10 +342,12 @@ function supportLegacyOptions(options: Object): PartialKPOptionsObject {
         level: 'warn',
         msg: `Path config.player.${propPath} will be deprecated soon. Please update your config structure as describe here: ${__CONFIG_DOCS_URL__}`
       });
-      const propValue = Utils.Object.getPropertyPath(options, propPath);
-      const propObj = Utils.Object.createPropertyPath({}, targetPath, propValue);
-      Utils.Object.mergeDeep(options, propObj);
-      Utils.Object.deletePropertyPath(options, propPath);
+      if (!Utils.Object.hasPropertyPath(options, targetPath)) {
```
Comment (yairans) [inline src/common/utils/setup-helpers.js]: update `supportLegacyOptions` test to cover your change

File: src/common/utils/setup-helpers.js [hunk 4]
```diff
@@ -318,6 +319,24 @@ function checkNativeTextTracksSupport(options: KPOptionsObject): void {
   }
 }
 
+/**
+ * Sets config option fullscreen element for Vr Mode support
+ * @private
+ * @param {KPOptionsObject} options - kaltura player options
+ * @returns {void}
+ */
+function configureVrRelevantOptions(options: KPOptionsObject): void {
```
Comment (OrenMe) [inline src/common/utils/setup-helpers.js]: I would say `Relevant` is redundant as any config is relevant per-se.

File: src/common/utils/setup-helpers.js [hunk 5]
```diff
@@ -318,6 +319,24 @@ function checkNativeTextTracksSupport(options: KPOptionsObject): void {
...
+  if (options.plugins.vr && !options.plugins.vr.disable && !options.ui.disable) {
```
Comment (yairans) [inline src/common/utils/setup-helpers.js]: Why `inBrowserFullscreen` depending on ui?

Comment (Yuvalke) [reply src/common/utils/setup-helpers.js]: It's not depend, it depends for default value, there isn't reason to set default inbrowserfullscreen for ui disable as was before

File: src/common/utils/setup-helpers.js (hunk 4 above)
Comment (Yuvalke) [reply src/common/utils/setup-helpers.js]: I'll change it to default instead

Comment (Yuvalke) [reply src/common/utils/setup-helpers.js]: I'll change it `to configureVrDefaultOptions ` instead

**Pull request merged**